lib/Parser/Scan.cpp

```diff
@@ -668,7 +669,11 @@ typename Scanner<EncodingPolicy>::EncodedCharPtr Scanner<EncodingPolicy>::FScanN
     else
     {
 LFloat:
-        *pdbl = Js::NumberUtilities::StrToDbl(p, &pchT, likelyInt);
+        if (m_scriptContext->GetConfig()->IsESBigIntEnabled())
+        {
+            likelyType = LikelyNumberType::BigInt;
```

Comment:
Why is this the default likely type?

Reply:
This is a hack, I need to use this arg in other function to say that ESBigInt is enabled

Reply:
A couple of people have commented on this now, so I'd recommend reconsidering the design. Passing a LikelyNumberType as a flag that enables BigInt is a violation of the principle of least astonishment. I think adding another parameter is fine if it simplifies the API. If there are lots of callers you don't want to update, default the new parameter to false so you don't have to go update them all.

Reply:
Or at the very least we need a big block comment above the definition of StrToDbl explaining how the params work.